src/blockstore/fs.ts

```diff
@@ -0,0 +1,47 @@
+import fs from 'fs'
+import os from 'os'
+
+import { CID } from 'multiformats'
+import { Block } from '@ipld/car/api'
+
+export class FsBlockStore {
+  store: Set<string>
+
+  constructor() {
+    this.store = new Set()
+  }
+
+  async put({ cid, bytes }: { cid: CID, bytes: Uint8Array }) {
+    const cidStr = cid.toString()
+    const location = `${os.tmpdir()}/${cidStr}`
+    await fs.promises.writeFile(location, bytes)
+
+    this.store.add(cidStr)
+
+    return { cid, bytes }
+  }
+
+  async get(cid: CID): Promise<Block> {
+    const cidStr = cid.toString()
+    const location = `${os.tmpdir()}/${cidStr}`
+    const bytes = await fs.promises.readFile(location)
+
+    return { cid, bytes }
+  }
+
+  async * blocks() {
+    for (const cidStr of this.store) {
+      const location = `${os.tmpdir()}/${cidStr}`
+      const bytes = await fs.promises.readFile(location)
+
+      yield { cid: CID.parse(cidStr), bytes }
+    }
+  }
+
+  async close () {
+    for (const cidStr of this.store) {
+      const location = `${os.tmpdir()}/${cidStr}`
+      await fs.promises.rm(location)
```

Comment:
Yeah, I was thinking about this when I did the close/destroy function. I actually wanted to discuss this further, as I am not sure removing everything from the Blockstore to be the expected behaviour?

While we will use this as a temporary storage, the user could still want to keep it (for instance for providing?).

Reply:
Agree, i was looking at using [tempy](https://github.com/sindresorhus/tempy) + [datastore-fs](https://github.com/ipfs/js-datastore-fs). Creating a datastore-fs instance with a random temp dir might be the way to go.

@alanshaw the existing datastore impls were more complicated than i expected, i assume to deal with sharding things in fs land to avoid creating too many dirs in on dir. Are there any other things that mean we should look to the existing datastore-* instead of rolling our own here? the ipfs-car usecase is somewhat simpler, but I don't want to miss some gotcha that means we should have just used the existing things.

Reply:
@olizilla , just to make sure, you expect the provided blockstore to be erased in the end?

I was thinking more on the way that a user may want to have the same blockstore used when getting car files, and could provide the blocks to the network. But, I might be thinking ahead of our current needs and probably we can destroy it and make it clear in the docs